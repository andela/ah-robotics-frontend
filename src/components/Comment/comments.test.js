import { shallow } from 'enzyme';
import React from 'react';
import AddCommentComponent from './addComment';
import CommentsListComponent from './commentList';
import SingleCommentComponent from './comment';
import UpdateCommentComponent from './updateComment';


describe('Add Comment Component Tests', () => {
  const wrapper = shallow(<AddCommentComponent />);

  it('app should render AddCommentComponent', () => {
    expect(wrapper.length).toBe(1);
  });

  it('it should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Comment List Component Tests', () => {
  const user = {};
  const wrapper = shallow(<CommentsListComponent
    user={user}
  />);

  it('app should render CommentsListComponent', () => {
    expect(wrapper.length).toBe(1);
  });

  it('it should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Single Comment Component Tests', () => {
  const props = {
    author: {
      image: 'adasdas',
    },
  };

  const user = {};

  const wrapper = shallow(<SingleCommentComponent
    comment={props}
    user={user}
  />);

  it('app should render SingleCommentComponent', () => {
    expect(wrapper.length).toBe(1);
  });

  it('it should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Update Comment Component Tests', () => {
  const commentId = 1;
  const commentDetails = { body: 'hello world' };
  const wrapper = shallow(
    <UpdateCommentComponent
      commentId={commentId}
      commentDetails={commentDetails}
    />,
);

  it('app should render UpdateCommentComponent', () => {
    expect(wrapper.length).toBe(1);
  });

  it('it should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
