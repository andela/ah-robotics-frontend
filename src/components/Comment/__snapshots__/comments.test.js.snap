// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Comment Component Tests it should match snapshot 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`Comment List Component Tests it should match snapshot 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`Single Comment Component Tests it should match snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Update Comment Component Tests it should match snapshot 1`] = `
<Route>
  <Component />
</Route>
`;
